bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/wizards/AbstractWizardRegistry.java

```diff
@@ -69,7 +73,15 @@ public IWizardDescriptor findWizard(String id) {
 	@Override
 	public IWizardDescriptor[] getPrimaryWizards() {
 		initialize();
-		return primaryWizards;
+
```

Comment:
I don't think this is the right way to do so.
`getPrimaryWizards()` is called from the `AbstractExtensionWizardRegistry` methods that add/remove wizards and so filtering here would break that. I believe this has to be solved not here, but in "upper" layer where UI contributions are done.

What is about `org.eclipse.ui.internal.dialogs.NewWizardNewPage` ? There is already a filter for wizards  - `org.eclipse.ui.internal.dialogs.NewWizardNewPage.filter` . Why does it not work for PDE project wizard? I believe the fix there would make more sense.